source/newtonian/three_dimensional/OpticalDepthCalc.cpp

```diff
@@ -50,7 +50,7 @@ namespace
 		if (surface_density.at(index - 1).second > surface_density.at(index + 1).second)
 		{
 			double total_density = surface_density[index - 1].second + density_self;
-			for (size_t i = index - 1; i >= 0; --i)
+			for (size_t i = index - 1; i > 0; --i)
```

Comment:
Why was this changed? This changes the loop count
@bolverk

Reply:
I think there is a bug in the loop count, I think it is best to change the loop variable to be an int